File: CollaborationFront/WebContent/c_job/JobController.js
"use strict"

app.controller('JobController',['JobServices','$http','$q','$rootScope','$location','$scope',function($http,$q,$location,$scope,$rootScope)
{
	var self = this;
	
	self.job = {id : '', tittle : '', qualification : '', description : '', status : '', date : ''};
	
	self.jobs = [];
	
	self.fetchAllJobs = function()
	{
		JobServices.fetchAllJobs().then
		(
				function(d)
				{
					self.jobs=d;
				},
				function(errResponse)
				{
					console.log("Error While Fetching all Jobs !!,.,!!,.,.")
				}
		)
	},
	
	
	self.jobByID = function(id)
	{
			JobServices.jobById(id).then
			(
					function(d)
					{
						self.job = d;
					},
					function(errRespose)
					{
						console.log("Error While Fetching Jon ,.,!!,.,!!,.,")
					}
			)
	},
	
	
	self.createJob = function(job)
	{
		JobServices.createJob(job).then
		(
				function(d)
				{
					alert("Thankyou Job has been Posted Successfull")
				},
				function(errResponse)
				{
					console.log("Error while posting job,, please try again after sometime ,.,!!,.,!!,.,")
				}
		)
	}
	
	self.updateJob = function(job)
	{
		JobServices.updateJob(job).then
		(
				function(d)
				{
					alert("Thankyou Job has been udpate Successfully")
				},
				function(errResponse)
				{
					console.log("Error While Updating Job.,.,Please try again after sometime,.,!!,.,!!,.,")
				}
		)
	}
}])

File: collaborationback/src/main/java/com/niit/collaborationbackend/controller/JobController.java
package com.niit.collaborationbackend.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import com.niit.collaborationbackend.DAOIMPL.JobDAOIMPL;
import com.niit.collaborationbackend.model.Job;

@Component
public class JobController {

	@Autowired
	Job job;
	
	@Autowired
	JobDAOIMPL jobDAO;
	
	
	
	@RequestMapping("/allJobs")
	public ResponseEntity<List<Job>> getAllJobs(){
		
		List<Job> jobs = jobDAO.list();
		if(jobs==null){
			job.setErrorCode("404");
			job.setErrorMessage("No Jobs Were found");
			jobs.add(job);
		}
		
		return new ResponseEntity<List<Job>>(jobs,HttpStatus.OK);
	}
	
	@RequestMapping("/jobByID/{id}")
	public ResponseEntity<Job> getJobById(@PathVariable("id") String jobId)
	{
		Job job = jobDAO.get(jobId);
		if(job==null)
		{
			job= new Job();
			job.setErrorCode("404");
			job.setErrorMessage("Error while getting job by Id = "+jobId);
		}
		return new ResponseEntity<Job>(job, HttpStatus.OK);
	}
	
	@RequestMapping("/savejob")
	public ResponseEntity<Job> saveJob(@RequestBody Job job)
	{
		if(jobDAO.save(job)==false)
		{
			job.setErrorCode("404");
			job.setErrorMessage("Error while saving Job,., Please try again after sometime,.,!!,.,!!,.");
		}
		else
		{
			job.setErrorCode("200");
			job.setErrorMessage("Thankyou!! Job has beed posted Successfully,.,!!.,.!!!,.,");
		}
		return new ResponseEntity<Job>(job,HttpStatus.OK);
	}
	
	@RequestMapping("/updatejob")
	public ResponseEntity<Job> updateJob(@RequestBody Job job){
		if(jobDAO.update(job)==false){
			job.setErrorCode("404");
			job.setErrorMessage("Error while update Job,., Please try again after sometime,.,!!,.,!!,.,");
		}
		else
		{
			job.setErrorCode("200");
			job.setErrorMessage("Thankyou!! Job has been update Successfully,.,!!,.,!!,.,");
		}
		
		return new ResponseEntity<Job>(job,HttpStatus.OK);
	}
	
}